### PR TITLE
Test instance on steps

### PR DESCRIPTION
**Pull request by koterpillar**

### CODE REVIEW

File: aloe/testclass.py [hunk 1]
```diff
@@ -48,6 +48,12 @@ def testclass(self):
         """
         return self.feature.testclass
 
+    test = None
+    """
+    The test currently running the step, or None if not currently in a test
+    (e.g. in a `before_feature` callback).
```
Comment (danni) [inline aloe/testclass.py:54]: Document they should use `testclass` in this case.

File: aloe/testclass.py [hunk 2]
```diff
@@ -295,7 +302,15 @@ def make_steps(cls, step_container, steps,
...
-            '    func{i}(step{i}, *args{i}, **kwargs{i})'.format(i=i)
+            # This has to be a single statement, in order to set its source
+            # location as a whole below
+            '\n'.join((
+                '    try:',
+                '        step{i}.test = self',
+                '        func{i}(step{i}, *args{i}, **kwargs{i})',
+                '    finally:',
+                '        step{i}.test = None',
+            )).format(i=i)
```
Comment (danni) [inline aloe/testclass.py]: What's wrong with the good old Python multiline string?

File: docs/classes.rst [hunk 3]
```diff
@@ -64,9 +64,16 @@ Step
 
         The :class:`Background` this step belongs to (if inside a background).
 
+    .. attribute:: test
+
+        The instance of :class:`unittest.TestCase` running the current test,
+        or None if not currently in a test (e.g. in a
+        :func:`~aloe.before.each_feature` callback).
```
Comment (danni) [inline docs/classes.rst:71]: Document that you can still use `testclass` in this case.

**Comment by koterpillar** [comment]
Fixes #84.

File: docs/classes.rst (hunk 3 above)
Comment (koterpillar) [inline docs/classes.rst:71]: Don't think that's needed - the only way to get to steps in this case is through the feature, which has the reference itself.

**Pull request merged**